src/Propel/Generator/Platform/MysqlPlatform.php

```diff
@@ -952,6 +952,26 @@ public function doQuoting($text)
         return '`' . strtr($text, ['.' => '`.`']) . '`';
     }
 
+    /**
+     * Get the default On Delete behavior for foreign keys when not explicity set.
+     *
+     * @return string
+     */
+    public function getDefaultForeignKeyOnDeleteBehavior()
+    {
+        return ForeignKey::RESTRICT;
```

Comment:
It seems like the docs for version 8 are incorrect. I am running MySQL Community Server 8.0.25 (the latest version) and when I execute a CREATE TABLE statement with a foreign key without specifying the ON DELETE or ON UPDATE behavior, it sets it to RESTRICT, not NO ACTION. I have actually implemented the distinction between version 5 and 8 as you suggested but when I test it on my database, the bug that I am trying to solve is still there because of the behavior I just mentioned. So I will revert that "compatibility" fix.

Reply:
I did an extra test where I now executed a CREATE TABLE statement with a foreign key with an explicit ON DELETE behavior set to NO ACTION. Turns out that this is just created as a RESTRICT behavior. So this does not seem to be a problem as specifying the NO ACTION behavior is not supported according to the Propel XML schema docs (http://propelorm.org/documentation/reference/schema.html#foreign-key-element).

Reply:
Hmm, that's weird. I am running MySql server version 8.0.25-0ubuntu0.20.04.1, and it behaves exactly as specified in the docs:
```sql
create table t(
	id1 int,
    foreign key (id1) references book(id) on delete no action on update no action,
    foreign key (id1) references book(id) on delete restrict on update restrict,
    foreign key (id1) references book(id) on delete set null on update set null,
    foreign key (id1) references book(id) on delete cascade on update cascade
);
show create table t;
```
gives
```
'CREATE TABLE `t` (
  `id1` int DEFAULT NULL,
  KEY `id1` (`id1`),
  CONSTRAINT `t_ibfk_1` FOREIGN KEY (`id1`) REFERENCES `book` (`id`),
  CONSTRAINT `t_ibfk_2` FOREIGN KEY (`id1`) REFERENCES `book` (`id`) ON DELETE RESTRICT ON UPDATE RESTRICT,
  CONSTRAINT `t_ibfk_3` FOREIGN KEY (`id1`) REFERENCES `book` (`id`) ON DELETE SET NULL ON UPDATE SET NULL,
  CONSTRAINT `t_ibfk_4` FOREIGN KEY (`id1`) REFERENCES `book` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci'
```
So it does indeed omit the `NO ACTION`, and not `RESTRICT`. I suppose we cannot rely on version alone. Also, MariaDB uses the same platform, but with different versioning (and it seems to use `RESTRICT` consistently).

If you want to go through the extra trouble, you can easily adjust your code to be able to detect both options as default (i.e. by changing `getDefaultOnDeleteBehavior()` to `isDefaultOnDeleteBehavior($detectedBehavior))`. Considering how there is an easy workaround for the issue - not declaring the default behavior - I think this is ok then as it is.

Reply:
Oh, just for reference, the propel doc is misleading. You actually can use `onDelete="no action"`, there just is no alias like `setnull`.

Reply:
Wait, I think you misunderstood me. The behavior you are seeing is also what I am seeing. So if you create a FK with behavior RESTRICT, NO ACTION or without defining it, it actually sets RESTRICT. So, if the Propel XML declares either of those three options, it should match RESTRICT. Currently, it will only give a match if RESTRICT is set or if the behavior is not set (empty or undefined). So if we indeed want to support declaring `onDelete="no action"` then the implementation should be further extended. But then indeed I'm not sure whether that will also be compatible with MySQL 5 and MariaDB.  It depends on how they behave when a NO ACTION behavior is specified. If it also results in a RESTRICT, it should be fine. If not, the version should be checked first.

Reply:
Ok, just tested and it appears that MariaDB explicitly sets NO ACTION if that is specified. So that means that if we want to support the `onDelete="no action"` declaration in the XML that we will have to differentiate between MySQL and MariaDB at least OR just interpret both behaviors as equal (e.g. RESTRICT = NO ACTION, also for MariaDB). Or, we can leave it as is and indeed advise to omit it in the XML instead. Let me know which option you prefer so we can wrap this up.

Reply:
Ok, after some struggling with unit tests I think I managed to implement a solution that will solve all issues for both MySQL 5 and 8 and also for MariaDB.

The situation before this PR can be described as follows:
|db|xml|result in db|reverse parsed as|comparison|
|---|---|---|---|---|
|mysql5|null|null|null|match|
|mysql5|restrict|null|null|no match|
|mysql5|no action|no action|no action|match|
|mysql8|null|null|null|match|
|mysql8|restrict|restrict|null|no match|
|mysql8|no action|null|null|no match|
|mariadb|null|null|null|match|
|mariadb|restrict|null|null|no match|
|mariadb|no action|no action|no action|match|

This PR takes away a mapping that is done during the reverse parsing and adds a fallback to a DB version-specific default:
|db|xml|result in db|reverse parsed as|comparison|
|---|---|---|---|---|
|mysql5|null|null|null|match (null defaults to restrict)|
|mysql5|restrict|null|null|**match (null defaults to restrict)**|
|mysql5|no action|no action|no action|match|
|mysql8|null|null|null|match (null defaults to no action)|
|mysql8|restrict|restrict|**restrict**|**match**|
|mysql8|no action|null|null|**match (null defaults to no action)**|
|mariadb|null|null|null|match (null defaults to restrict)|
|mariadb|restrict|null|null|**match (null defaults to restrict)**|
|mariadb|no action|no action|no action|match|

Reply:
Okay no, I reverted this as I realized (after testing more intensively) that the schema that is created based on the XML to compare with the reversed schema from the actual database does not contain an actual database connection so it has no way to know which database version should be used for the XML, resulting in a null error when trying to get the database version attribute for the XML database. I honestly don't see a way to address this problem anymore :/

If you still have ideas or suggestions, please share but otherwise I would propose to leave the PR as-is for now, knowing that the NO ACTION behavior won't work correctly when using it with MySQL 8.

Reply:
Okay, sorry for the spamming but I think I finally found a permanent solution for the connection problem. So hopefully, this PR can be wrapped up now ;)